### PR TITLE
add group uuid as parameter for deviceview endpoint

### PR DESCRIPTION
**Pull request by acosferreira**
# Description
enable group_uuid as parameter to device view endpoint to be able to get list of device based on the insights group

FIXES: <THEEDGE-3581>

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

File: pkg/routes/devices.go [hunk 1]
```diff
@@ -91,6 +91,11 @@ var devicesFilters = common.ComposeFilters(
 		QueryParam: "image_id",
 		DBField:    "devices.image_id",
 	}),
+	// Filter handler for "groupUUID"
+	common.ContainFilterHandler(&common.Filter{
```
[the remark this answers is not in the document]
Comment (acosferreira) [reply pkg/routes/devices.go:95]: It's https://github.com/RedHatInsights/edge-api/pull/2266/files#diff-61e5cd45b4875331292acc0eeddea19a8fb7bb1fe2d7f25e6a9124ca71f4b7e0R22

**Review by ldjebran — COMMENTED** [review]
looks good, missing one unittest

File: pkg/routes/devices_test.go [hunk 2]
```diff
@@ -1070,3 +1070,91 @@ func TestGetDevicesViewWithinDevicesWitUUIDNotExist(t *testing.T) {
...
+
+	url := fmt.Sprintf("/deviceview?groupUUID=%v", groupUUID)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		Expect(err).ToNot(HaveOccurred())
+	}
+	rr := httptest.NewRecorder()
+
+	ctx := dependencies.ContextWithServices(req.Context(), mockServices)
+	req = req.WithContext(ctx)
```
Comment (ldjebran) [inline pkg/routes/devices_test.go:1157]:
```suggestion
	req = req.WithContext(ctx)
	router.ServeHTTP(rr, req)
```

**Review by adeliaferreira — APPROVED** [review]
Lgtm

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Pull request merged**